lib/ansible/modules/cloud/openstack/os_keystone_endpoint.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/python
+# Copyright (c) 2017 VEXXHOST, Inc.
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: os_keystone_endpoint
+short_description: Manage OpenStack Identity service endpoints
+extends_documentation_fragment: openstack
+author:
+    - Mohammed Naser (@mnaser)
+    - Alberto Murillo (@albertomurillo)
+version_added: "2.5"
+description:
+    - Create, update, or delete OpenStack Identity service endpoints. If a
+      service with the same combination of I(service), I(interface) and I(region)
+      exist, the I(url) and I(state) (C(present) or C(absent)) will be updated.
+options:
+   service:
+     description:
+        - Name or id of the service.
+     required: true
+   interface:
+     description:
+        - Interface of the service.
+     choices: [admin, public, internal]
+     required: true
+   url:
+     description:
+        - URL of the service.
+     required: true
+   region:
+     description:
+        - Region that the service belongs to. Note that I(region_name) is used for authentication.
+   enabled:
+     description:
+        - Is the service enabled.
+     default: True
+   state:
+     description:
+       - Should the resource be C(present) or C(absent).
+     choices: [present, absent]
+     default: present
+requirements:
+    - "shade>=1.11.0"
+'''
+
+EXAMPLES = '''
+- name: Create a service for glance
+  os_keystone_endpoint:
+     cloud: mycloud
+     state: present
+     service: glance
+     interface: public
+     url: http://controller:9292
+     region: RegionOne
+
+- name: Delete a service
+  os_keystone_endpoint:
+     cloud: mycloud
+     state: absent
+     name: image
```

Comment:
should this be 'service: image'

Reply:
its actually service name or id so `service: glance`

Reply:
fixed